### ISSUE TITLE
New Shows not recognized

### ISSUE THREAD

**Issue by A69Tor** [m0ngr31/kodi-alexa#66]
As mentioned with [#40](https://github.com/m0ngr31/kodi-alexa/issues/40), new shows are not being detected correctly prior to adding to the interaction model.  Ran across this issue again with a new show that does not have a number in the title.

The show is titled "Letterkenny", which when requesting to play via the skill is continuously playing different shows like "Leverage" or "Luther".  When attempting to wikipedia Letterkenny via Alexa there is no issue what so ever with the recognition of the word.

Upon addition of "Letterkenny" to the interaction model of the skill there is no longer any issue recognizing the title of the show.

**Comment by jingai** [comment]
I just tried it with a movie I hadn't added to the slots and it works fine here.

I'm guessing it just wasn't able to match "Letterkenny".  What did Alexa hear?

**Comment by A69Tor** [comment]
Alexa is not showing what it actually heard, it is showing the skill response of "Playing next unwatched episode of [Incorrect show]".  Again tested this with Letterkenny with the wikipedia skill and it heard it fine.  Tested it again after adding Letterkenny to the interaction model and it worked fine.  Tested it 10+ times prior to adding to the interaction model and it played the incorrect show every single time.

**Comment by jingai** [comment]
Click on More on the card to see what she actually heard.

I'm going to guess Letterkenny is going to get translated by her to something odd and if the fuzzy matching routines can't get close enough to Letterkenny, your solution for such problematic shows is to add them to the Slot.

**Comment by A69Tor** [comment]
Shows as Alexa hearing the incorrect show, just seems very odd that the first time after I add it to the slot it hears it without issue, and that with other skills it heard it correctly as well.

**Comment by jingai** [comment]
I need to know what she actually heard (before you added the show to the Slot) if you want help.

Please find that Card, click More, and copy and paste the text after "Alexa heard:" under Voice Feedback.

It isn't odd that it works after adding it to the slot.  What's happening is that without the entry in the Slot, Alexa is hearing something else that may sound a bit like "Letterkenny."  This is then passed on to the Kodi skill verbatim, where it then goes through a few ways of matching what it heard to your library.

If Alexa heard something like "Let her coney," "Leather kidney," etc etc, it may not match, or it may match to something else more strongly.

When you add it to the Slot, she matches it exactly to "Letterkenny" which is then passed to the Skill, where it obviously has no trouble matching that exactly.

In a nutshell: the matching isn't 100% perfect.  It's really good, IMHO, but when you encounter an exception that doesn't want to match, that's exactly what the Slots are for:  exceptions.

**Comment by A69Tor** [comment]
The two "What Alexa Heard" for this across all 10+ tests prior to adding the name to the slot were:

Alexa Heard: 'tell kodi play the next episode of luther'

and

Alexa Heard: 'tell kodi to play the next episode of leverage'

**Comment by jingai** [comment]
So my guess is that the skill matched "Luther" and "Leverage" more strongly to another show you have. 

The solution here is to add it to the Slot.

Amazon [no longer supports the AMAZON.LITERAL type](https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/docs/alexa-skills-kit-interaction-model-reference#literal-slot-type-reference), so there isn't much we can do about her converting what you said.

**Comment by A69Tor** [comment]
That being the case, would there be a way to possibly auto-update the slot when new shows/movies are added?

**Comment by jingai** [comment]
Unless Amazon has some API for modifying the models via a script, I don't see how. 

But, this shouldn't be something that occurs frequently anyway. You just picked a particularly odd one here :)

**Comment by A69Tor** [comment]
Alright sounds good! Thanks for the insight!
